### PR TITLE
Build RPM packages with Docker

### PR DESCRIPTION
**Pull request by astj**
related: #325 

Though we can build current rpm in travis directly, we cannot build systemd-based one (under development on #317).
So I'd like to introduce docker based rpm build procedure.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
-sudo: false
+sudo: true
```
Comment (haya14busa) [inline .travis.yml]: Isn't it better to use `sudo: required`? https://docs.travis-ci.com/user/docker/

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+services:
+- docker
 language: go
 go:
 - 1.7
 env:
   global:
   - PATH=/home/travis/gopath/bin:$PATH DEBIAN_FRONTEND=noninteractive
   - secure: "B34YsRebWxyx/UAHNAL5YWOgx2RE13TVfDC8FCCc7gAlogJcyAcPnC88fe8ECChHQC3GNnsHSi+GqgxTxzfoVkV+LUAFgHWbcF5O4O/4qHztKj7//72RwO+gXVr4U1Laz45bBr2r68XW0llj5tIlaEYxROJNLSAVDE4yDzmAHgc="
-sudo: false
```
Comment (haya14busa) [inline .travis.yml]: :innocent:

File: Makefile [hunk 3]
```diff
@@ -65,38 +65,46 @@ crossbuild-package-stage:
 	mv build/mackerel-agent-stage build-linux-386/
 
 rpm: crossbuild-package
+	mkdir -p rpmbuild
 	MACKEREL_AGENT_NAME=$(MACKEREL_AGENT_NAME) _tools/packaging/prepare-rpm-build.sh
-	rpmbuild --define "_sourcedir `pwd`/packaging/rpm-build/src" --define "_builddir `pwd`/build-linux-386" \
-	      --define "_version ${CURRENT_VERSION}" --define "buildarch noarch" \
-				-bb packaging/rpm-build/$(MACKEREL_AGENT_NAME).spec
+	docker run --rm -v "$(PWD)":/workspace -v "$(PWD)/rpmbuild":/rpmbuild astj/mackerel-rpm-builder:c6 \
```
Comment (haya14busa) [inline Makefile]: I guess it's better to remove `--rm` here because L75 uses same docker image.

Comment (astj) [inline Makefile]: `--rm` is not for images but for containers.  We cannot reuse this container in L75 by removing `--rm`. (It just causes stopped container remaining)

Comment (haya14busa) [inline Makefile]: Hmm.. I misunderstood it. sorry & thanks!
https://docs.docker.com/engine/reference/run/#/clean-up---rm

File: .travis.yml (hunk 1 above)
Comment (astj) [inline .travis.yml]: Maybe better. (Even `sudo: true` looks working)
Fixed in https://github.com/mackerelio/mackerel-agent/pull/330/commits/e018b7a40494dd7f10254f99f65ba5a9d342734d.

Build details of https://github.com/mackerelio/mackerel-agent/pull/330/commits/5c8d2e21dbe8f809df4a3ef16c974be98274dc60 shows this branch can build rpms with Docker. (Building rpms in `after-script` is re-reverted on https://github.com/mackerelio/mackerel-agent/pull/330/commits/e7055122568d0adca7be2331da9c515d5ac0d3e7)

**Comment by itchyny** [comment]
LGTM

**Comment by astj** [comment]
Thanks!

**Pull request merged**